test/helpers/position_assertions.cc

```diff
@@ -1187,44 +1205,58 @@ void SymbolSearchAssertion::checkAllForQuery(std::string query,
         auto &location = symbol->location;
         auto line = fmt::format("{}:{:{}}", location->uri, location->range->start->line, lineWidth);
         symbolsByLine[line].push_back(symbol);
+        assertionsByLine[line]; // create entry in assertionsByLine, if none exists yet.
     }
 
-    bool success = true;
-    // Show any expected symbols that weren't returned by LSP
+    int numUnfulfilled = 0;
+    int numUnexpected = 0;
+    // Look line-by-line for unmatched symbols and assertions
     for (auto entry : assertionsByLine) {
         auto lineInfo = entry.first;
-        auto expected = entry.second;
-        auto &actual = symbolsByLine[lineInfo];
-        if (expected.size() == actual.size()) {
-            continue;
+        auto assertionsOnLine = entry.second;
+        auto &symbolsOnLine = symbolsByLine[lineInfo];
+
+        // Report assertions with no matching symbol
+        for (auto &assertion : assertionsOnLine) {
+            bool foundMatchingSymbol = false;
+            for (auto &symbol : symbolsOnLine) {
+                if (assertion->matches(uriPrefix, symbol)) {
+                    foundMatchingSymbol = true;
```

Comment:
Why do you need this variable? It seems like you could just `continue` here.

Reply:
Continuing here would continue from the *inner* for loop on line 1222, whereas the continue needs to happen from the *outer* for loop on line 1220.

Reply:
Oh! Right; C++ doesn't support named loops, and we probably don't want to use `goto` (which is the most convenient alternative). Good call.